### PR TITLE
fix configure

### PR DESCRIPTION
**Pull request by jbowler**
- **Create the branch 'libpng18' for libpng version 1.8.0 alpha**
- **doc: Rename README, LICENSE, etc., to README.md, LICENSE.md, etc.**
- **doc: Move all manuals to the manuals/ directory**
- **build: Run `./configure && make maintainer-clean`; update .gitignore**
- **build: Raise the minimum required CMake version to 3.14**
- **cmake: Remove architecture-specific checks for hardware optimization**
- **Full compile time support for target-specific code**
- **MIPS support clean-up**
- **[libpng18] SIMD makefile updates**
- **ARM Ltd palette SIMD code**
- **Use standard C attributes**
- **[libpng18] fix autoconf**

### CODE REVIEW

**Comment by jbowler** [comment]
@ctruta: there is only one commit here.  All the reset is because of the prior unmerged changes.  This is all based on [libpng18].  Fixes #591, adds a .vimrc so those vile VIM users won't mis-indent the source (believe me I don't use 3-space softtabs unless I'm working on libpng), updates the .gitignore for the new maintainer mode system.

TBD: I'll force push a removal of the problematic macro in configure.ac... sorry I just forgot about it.

File: .vimrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+set shiftwidth=3
```
Comment (ctruta) [inline .vimrc:1]: I added the .editorconfig files for the users of all text editors that recognize the EditorConfig specification (https://editorconfig.org).

As a matter of fact, an EditorConfig plugin for Vim comes standard:
https://github.com/editorconfig/editorconfig-vim

I, too, am using it with Visual Studio Code (for coding) and with the `editorconfig-checker` program (for linting). I haven't used it with Vim, as I'm not that much of a Vim user to begin with, but if it doesn't work well enough for you, see if you can tweak the `.editorconfig` file until Vim does what you want.

Comment (jbowler) [reply .vimrc:1]: My vi was using it automatically but it didn't do 'set shiftwidth=3', so didn't even know it was there; my default shiftwidth is 4 so every time I editted a file the indentation ended up wrong.  #594 fixes this.  I'll close this PR.

**Comment by jbowler** [comment]
Effectively fixed.  #594 fixes the missing indent spacing in .editorconfig.

**Pull request closed without merging**